lib/spack/docs/packaging_guide.rst

```diff
@@ -2399,6 +2413,56 @@ the install locations of dependencies, or when you need to do something differen
 depending on the version, compiler, dependencies, etc. that your package is
 built with.  These parameters give you access to this type of information.
 
+^^^^^^^^^^^^^^^^^
+Adding new phases
+^^^^^^^^^^^^^^^^^
+
+Phases can be added before or after existing phases by using the ``run_before``
```

Comment:
That's not really correct. The *phases* are specified in the base build-system class, but you can run arbitrary code using the `@run_before(phase)` and `@run_after(phase)` decorator.

Reply:
For instance, every `CMakePackage` has 3 phases:

https://github.com/LLNL/spack/blob/9aafe21b6328d5944469b7a0933e5a446133f876/lib/spack/spack/build_systems/cmake.py#L65-L66

but any package can attach code in between.

Reply:
I've updated the wording. Have a look!